src/Microsoft.OpenApi/Services/OpenApiFilterService.cs

```diff
@@ -46,16 +56,55 @@ public static class OpenApiFilterService
                 {
                     var regex = new Regex(tagsArray[0]);
 
-                    predicate = (o) => o.Tags.Any(t => regex.IsMatch(t.Name));
+                    predicate = (url, operationType, operation) => operation.Tags.Any(tag => regex.IsMatch(tag.Name));
                 }
                 else
                 {
-                    predicate = (o) => o.Tags.Any(t => tagsArray.Contains(t.Name));
+                    predicate = (url, operationType, operation) => operation.Tags.Any(tag => tagsArray.Contains(tag.Name));
                 }
             }
+            else if (requestUrls != null)
+            {
+                var operationTypes = new List<string>();
+
+                if (source != null)
+                {
+                    var apiVersion = source.Info.Version;
+
+                    var sources = new Dictionary<string, OpenApiDocument> {{ apiVersion, source}};
+                    var rootNode = CreateOpenApiUrlTreeNode(sources);
+
+                    //Iterate through urls dictionary and fetch operations for each url
+                    foreach (var path in requestUrls)
+                    {
+                        var serverList = source.Servers;
+                        var url = FormatUrlString(path.Key, serverList);
+
+                        var openApiOperations = GetOpenApiOperations(rootNode, url, apiVersion);
+                        if (openApiOperations == null)
+                        {
+                            continue;
+                        }
+
+                        foreach (var ops in openApiOperations)
+                        {
+                            operationTypes.Add(ops.Key + url);
```

Comment:
We should only add the available operations if they are listed in the Postman collection.  I believe the list is in path.Value.

Reply:
Thanks for pointing that out. Resolving..